### PR TITLE
Load all pages of Facebook friends

### PR DESCRIPTION
**Pull request by dborkan**
Fixes https://github.com/freedomjs/freedom-social-firebase/issues/11

Tested in Chrome (by changing LIMIT to 2 and verifying that it successfully made the recursive calls to get all pages of friends).

### CODE REVIEW

File: src/facebook-social-provider.js [hunk 1]
```diff
@@ -67,21 +67,40 @@ FacebookSocialProvider.prototype.getOAuthTokenInteractive_ =
...
+
+  var LIMIT = 100;
+  if (!url) {
+    url = 'https://graph.facebook.com/v2.1/me/friends' +
+      '?access_token=' + this.loginState_.authData.facebook.accessToken +
+      '&format=json&redirect=false&limit=' + LIMIT;
+  }
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url);
+  var thisFacebookSocialProvider = this;
```
Comment (salomegeo) [inline src/facebook-social-provider.js:85]: Why do you need this?

**Comment by salomegeo** [comment]
What does it mean to change the LIMIT to 2? could we have changed the LIMIT to 5000 and that would've fixed it?

File: src/facebook-social-provider.js (hunk 1 above)
Comment (dborkan) [inline src/facebook-social-provider.js:85]: thisFacebookSocialProvider is used within the xhr.onload callback.  Within that callback, just using `this` won't give us what we want, and if we do the usual `.bind(this)` we won't be able to easily access this.response returned by the XHR

**Comment by dborkan** [comment]
I have about 10 FB friends who use uProxy.  Changing the LIMIT to 2 for testing means that it ensures that the loadContacts_ method is called recursively.  If I used LIMIT=100, all the contacts would be loaded after the first XHR request.  We could change the limit to 5000 but at some point it could be slow, and it's not clear what limits might exist on Facebook's side.  For now 100 seems like a safe option.

**Comment by dborkan** [comment]
Closing this - we no longer load the users Facebook friends in freedom-social-firebase, but instead friends are added via invites

**Pull request closed without merging**